src/lib/protocols/connections/ConnectionService.ts

```diff
@@ -106,19 +107,9 @@ class ConnectionService {
 
   async createConnection(): Promise<Connection> {
     const [did, verkey] = await this.context.wallet.createDid();
-    const did_doc = {
-      '@context': 'https://w3id.org/did/v1',
-      service: [
-        {
-          id: 'did:example:123456789abcdefghi#did-communication',
-          type: 'did-communication',
-          priority: 0,
-          recipientKeys: [verkey],
-          routingKeys: this.getRoutingKeys(),
-          serviceEndpoint: this.getEndpoint(),
-        },
-      ],
-    };
+    const publicKey = new PublicKey(`${did}#1`, PublicKeyType.ED25519_SIG_2018, did, verkey, true);
```

Comment:
What if we use `Authentication` object and pass a `publicKey` to it instead of hiding this information into `PublicKey`. Like this:

```
const publicKey = new PublicKey(`${did}#1`, PublicKeyType.ED25519_SIG_2018, did, verkey)
const authentication = new Authentication(publicKey)
const didDoc = new DidDoc(did, authentication, [publicKey], [service]);
```

This would allow us to get rid of the last flag argument and also align more with didDoc spec, where you have `publicKey` and `authentication` attributes on the same level.